exchanges/gateio/gateio_types.go

```diff
@@ -25,526 +54,2548 @@ var (
+	Fee          float64   `json:"fee,string"`
+}
+
+// OptionsTicker represents  tickers of options contracts
+type OptionsTicker struct {
+	Name                  string  `json:"name"`
+	LastPrice             float64 `json:"last_price"`
+	MarkPrice             float64 `json:"mark_price"`
+	PositionSize          float64 `json:"position_size"`
+	Ask1Size              float64 `json:"ask1_size"`
+	Ask1Price             float64 `json:"ask1_price,string"`
+	Bid1Size              float64 `json:"bid1_size"`
+	Bid1Price             float64 `json:"bid1_price,string"`
+	Vega                  string  `json:"vega"`
+	Theta                 string  `json:"theta"`
+	Rho                   string  `json:"rho"`
+	Gamma                 string  `json:"gamma"`
+	Delta                 string  `json:"delta"`
+	MarkImpliedVolatility string  `json:"mark_iv"`
+	BidImpliedVolatility  string  `json:"bid_iv"`
+	AskImpliedVolatility  string  `json:"ask_iv"`
+	Leverage              string  `json:"leverage"`
+
+	// Added fields for the websocket
+	IndexPrice string `json:"index_price"`
+}
+
+// OptionsUnderlyingTicker represents underlying ticker
+type OptionsUnderlyingTicker struct {
+	TradePut   float64 `json:"trade_put"`
+	TradeCall  float64 `json:"trade_call"`
+	IndexPrice float64 `json:"index_price,string"`
+}
+
+// OptionAccount represents option account.
+type OptionAccount struct {
+	User          int64   `json:"user"`
+	Currency      string  `json:"currency"`
+	ShortEnabled  bool    `json:"short_enabled"`
+	Total         float64 `json:"total,string"`
+	UnrealisedPnl string  `json:"unrealised_pnl"`
+	InitMargin    string  `json:"init_margin"`
+	MaintMargin   string  `json:"maint_margin"`
+	OrderMargin   string  `json:"order_margin"`
+	Available     float64 `json:"available,string"`
+	Point         string  `json:"point"`
+}
+
+// AccountBook represents account changing history item
+type AccountBook struct {
+	ChangeTime    time.Time `json:"time"`
+	AccountChange float64   `json:"change,string"`
+	Balance       float64   `json:"balance,string"`
+	CustomText    string    `json:"text"`
+	ChangingType  string    `json:"type"`
+}
+
+// UsersPositionForUnderlying represents user's position for specified underlying.
+type UsersPositionForUnderlying struct {
+	User          int     `json:"user"`
+	Contract      string  `json:"contract"`
+	Size          int     `json:"size"`
+	EntryPrice    float64 `json:"entry_price,string"`
+	RealisedPnl   float64 `json:"realised_pnl,string"`
+	MarkPrice     float64 `json:"mark_price,string"`
+	UnrealisedPnl float64 `json:"unrealised_pnl,string"`
+	PendingOrders int     `json:"pending_orders"`
+	CloseOrder    struct {
+		ID    int    `json:"id"`
+		Price string `json:"price"`
+		IsLiq bool   `json:"is_liq"`
+	} `json:"close_order"`
+}
+
+// ContractClosePosition represents user's liquidation history
+type ContractClosePosition struct {
+	PositionCloseTime time.Time `json:"time"`
+	Pnl               string    `json:"pnl"`
```

Comment:
I will stop suggesting float64's for now but could you go over the file to update.